TeamCharles/initial-site
# 79-view-model-refactor :: Refactor Views to accept View Models

## Status

**READY**
## Documentation

[ ] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[X] This PR does **NOT** require documentation.
## Description

Update all methods that return a `View()` to accept a viewmodel instead: `View(model)`. Every ViewModel inherits from the BaseViewModel, which stores a list of all Users and the CurrentUser.
## Resolves Issue Number

Resolves #79 
## Deploy Notes

``` sh
git pull --prune
git checkout 79-view-model-refactor
dotnet run
```
## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
- Make sure you can retrieve every view
  - ProductTypes List
  - Products List
  - Product Detail
  - Add a Product
  - Add a User
  - Cart
## Impacted Files in Application

List general components of the application that this PR will affect:
- All Views
- All Controller Methods that return a View
- Created ViewModel folder with several new Model classes
